java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -499,461 +225,212 @@ public boolean equals(Object obj) {
-  public ByteBuf setIntLE(int index, int value) {
-    chk(index, 4);
-    PlatformDependent.putInt(addr(index), Integer.reverseBytes(value));
-    return this;
-  }
-
-  @Override
-  public ArrowBuf setLong(int index, long value) {
-    chk(index, 8);
-    PlatformDependent.putLong(addr(index), value);
-    return this;
-  }
-
-  /**
-   * Sets the specified 64-bit long integer at the specified absolute {@code index}
-   * in this buffer with Big Endian byte order.
-   */
-  @Override
-  public ByteBuf setLongLE(int index, long value) {
-    chk(index, 8);
-    PlatformDependent.putLong(addr(index), Long.reverseBytes(value));
-    return this;
+  public short getShort(int index) {
+    chk(index, 2);
+    return PlatformDependent.getShort(addr(index));
   }
 
-  @Override
-  public ArrowBuf setChar(int index, int value) {
+  public ArrowBuf setShort(int index, int value) {
     chk(index, 2);
     PlatformDependent.putShort(addr(index), (short) value);
     return this;
   }
 
-  @Override
-  public ArrowBuf setFloat(int index, float value) {
-    chk(index, 4);
-    PlatformDependent.putInt(addr(index), Float.floatToRawIntBits(value));
-    return this;
-  }
-
-  @Override
-  public ArrowBuf setDouble(int index, double value) {
-    chk(index, 8);
-    PlatformDependent.putLong(addr(index), Double.doubleToRawLongBits(value));
-    return this;
-  }
-
-  @Override
   public ArrowBuf writeShort(int value) {
     ensure(2);
     PlatformDependent.putShort(addr(writerIndex), (short) value);
     writerIndex += 2;
     return this;
   }
 
-  @Override
   public ArrowBuf writeInt(int value) {
     ensure(4);
     PlatformDependent.putInt(addr(writerIndex), value);
     writerIndex += 4;
     return this;
   }
 
-  @Override
   public ArrowBuf writeLong(long value) {
     ensure(8);
     PlatformDependent.putLong(addr(writerIndex), value);
     writerIndex += 8;
     return this;
   }
 
-  @Override
-  public ArrowBuf writeChar(int value) {
-    ensure(2);
-    PlatformDependent.putShort(addr(writerIndex), (short) value);
-    writerIndex += 2;
-    return this;
-  }
-
-  @Override
   public ArrowBuf writeFloat(float value) {
     ensure(4);
     PlatformDependent.putInt(addr(writerIndex), Float.floatToRawIntBits(value));
     writerIndex += 4;
     return this;
   }
 
-  @Override
   public ArrowBuf writeDouble(double value) {
     ensure(8);
     PlatformDependent.putLong(addr(writerIndex), Double.doubleToRawLongBits(value));
     writerIndex += 8;
     return this;
   }
 
-  @Override
-  public ArrowBuf getBytes(int index, byte[] dst, int dstIndex, int length) {
-    udle.getBytes(index + offset, dst, dstIndex, length);
-    return this;
-  }
-
-  @Override
-  public ArrowBuf getBytes(int index, ByteBuffer dst) {
-    udle.getBytes(index + offset, dst);
-    return this;
-  }
-
-  @Override
   public ArrowBuf setByte(int index, int value) {
     chk(index, 1);
     PlatformDependent.putByte(addr(index), (byte) value);
     return this;
   }
 
```

Comment:
Please try to add javadoc to each method

Reply:
Let's make sure people are happy with overall shape/approach of patch before we spend time on this type of changes.